### PR TITLE
Add ability to drop into a shell

### PR DESCRIPTION
**Pull request by felddy**
## 🗣 Description ##

Add the ability to pass `--shell` to the container and drop into a shell after the bucket is sync'd.

## 💭 Motivation and context ##

Some of our legacy systems are having trouble with the cross-signed ISRG Root X1 in the new LetsEncrypt certificate chain.  We've been unable to get the quoted magic to pass into the container and be accepted by certbot: `--preferred-chain "ISRG Root X1"`

This PR allows the user to drop to a shell in the container and issue commands directly to certbot, bypassing the transmutation of the arguments from the host environment.

See:
- https://www.openssl.org/blog/blog/2021/09/13/LetsEncryptRootCertExpire/
- https://community.letsencrypt.org/t/howto-obtain-a-full-certificate-chain-without-a-cross-signed-isrg-root-x1/161445

## 🧪 Testing ##

This container was tested with:
- local builds on my development machine as well as image artifacts created from the PR workflow:  `cisagov/certboto:improvement-drop-into-shell`
- by @dav3r to create the new certificates for the legacy systems.
- unit / system test jobs in continuous integration.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: src/entrypoint.sh
Comment (jsf9k) [inline src/entrypoint.sh]: Please explain what sort of checks you are disabling and why.

**Review by jsf9k — COMMENTED** [review]
I humbly submit one suggestion for your consideration.

File: src/entrypoint.sh
Comment (felddy) [reply src/entrypoint.sh]: See: 83ff65dee205178316fda619a01fde524ee0563b

Comment (felddy) [reply src/entrypoint.sh]: I discovered that I could clear up a bunch of the shellcheck complaints by correctly identifying the busybox shell as `dash`:

https://github.com/cisagov/certboto-docker/blob/b7353a8b645c5afd108f4264a75b60d1b1015d4f/src/entrypoint.sh#L1-L2

File: src/entrypoint.sh
Comment (dav3r) [inline src/entrypoint.sh:2]: Can you please add some comments here explaining the busybox and shellcheck lines?

**Review by dav3r — CHANGES_REQUESTED** [review]
This looks good, but I need to see some comment love before I can approve it.

File: src/entrypoint.sh
Comment (felddy) [reply src/entrypoint.sh:2]: See: a14dd7a
and: 🖕

Comment (dav3r) [reply src/entrypoint.sh:2]: Thanks!

**Review by dav3r — APPROVED** [review]
Approved with gusto!
🐚

File: Dockerfile
Comment (mcdonnnj) [inline Dockerfile]:
```suggestion
  pip==21.3.1 \
  setuptools==58.3.0  \
```

File: Dockerfile
Comment (mcdonnnj) [inline Dockerfile]:
```suggestion
  awscli==1.21.3 \
  boto3==1.19.3 \
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Makes sense. I would like the image version to be bumped since we are adding a new command line option. I also have two suggestions just to bump to the latest Python package versions.

**Comment by felddy** [comment]
Hurry before the dependencies update again, and this becomes a rolling PR!

- 736c58c49f15118a0e4abf559befb08edc979451
- 9f83da08c5359d29680beb9681bdf59cb84269f2

File: src/entrypoint.sh
Comment (jsf9k) [inline src/entrypoint.sh:34]: Is this still necessary, now that argument parsing is being done correctly?  I seem to recall that the unescapable thing that @dav3r was trying to pass started working once you added proper argument parsing.  My memory has been known to be faulty, though.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I have an important question to resolve before merging this PR.

File: src/entrypoint.sh
Comment (dav3r) [reply src/entrypoint.sh:34]: @felddy and I re-tested this with our troublesome quoted argument (`--preferred-chain "ISRG Root X1"`) and it still is not parsed correctly, even with the modified parsing in this PR.  We were not able to get it to work after a trying a variety of quote and space escape syntax options- you are welcome to try or suggest ideas here.  Until we figure that out, we still require the functionality offered by this new `--shell` flag.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
I'm still waiting on that Docker image version bump~

File: src/entrypoint.sh
Comment (jsf9k) [reply src/entrypoint.sh:34]: OK, I am suffering from a false memory or misunderstanding then (as usual).  I'll resolve this conversation.

**Review by jsf9k — APPROVED** [review]
Approved, once you bump the version as @mcdonnnj has requested.

File: src/entrypoint.sh
Comment (dav3r) [reply src/entrypoint.sh:34]: @felddy mentioned that he thought the modified argument parsing might have solved our issues, so that is probably what you are remembering.  It turned out to not be true once we actually tested it.

**Comment by dav3r** [comment]
This PR needs to get out the door- it is causing some headaches for us.  @felddy can you please get it whipped into shape?  If not, let me know and I will take it over.

**Review by dav3r — APPROVED** [review]
Still all good in the certboto hood! 🚀

File: README.md
Comment (jsf9k) [inline README.md]:
```suggestion
|`cisagov/certboto:0.1`| The most recent release matching the major and minor version numbers. |
```

**Review by jsf9k — APPROVED** [review]
Approved, but I think there is one minor tweak that needs to be made to the README.

**Review by mcdonnnj — APPROVED** [review]
:shipit: 🚀

**Pull request merged**
